src/composables/useTime.ts

```diff
@@ -24,6 +28,16 @@ export function toJsTimestamp(unixTimestamp: number): number {
   return unixTimestamp * oneSecondInMs;
 }
 
+export function getPreviousThursday(date: Date): Date {
+  const daysSinceThursday = (date.getDay() - 4) % 7;
```

Comment:
```suggestion
  const daysSinceThursday = (date.getDay() - 4 + 7) % 7;
```

This doesn't work for Mon-Wed currently as it doesn't handle negatives correctly so we need to bump up by 7

Reply:
Granted, I think this bug got added by copy-pasting my snippet.